src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/RazorSemanticTokensInfoService.cs

```diff
@@ -53,19 +54,27 @@ internal class RazorSemanticTokensInfoService : IRazorSemanticTokensInfoService
         var codeDocument = await documentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
 
         cancellationToken.ThrowIfCancellationRequested();
-        var razorSemanticRanges = TagHelperSemanticRangeVisitor.VisitAllNodes(codeDocument, range, razorSemanticTokensLegend);
+        List<SemanticRange>? razorSemanticRanges = null;
```

Comment:
Should we move these to pooled while we're here?

Reply:
I've changed GetCSharpSemanticRangesAsync to internally use a pooled list; I don't think we're likely to change the TagHelperVisitor.Visit's method signature to return a pooled list by signature, but it looks possible.   Thoughts?